app/controllers/users_controller.rb

```diff
@@ -168,6 +168,11 @@ def profile
         @twitter = @profile_user.social_link("twitter")
         @facebook = @profile_user.social_link("facebook")
         @instagram = @profile_user.social_link("instagram")
+        @github_user = @profile_user.get_value_of_power_tag("github")
+        @twitter_user = @profile_user.get_value_of_power_tag("twitter")
+        @facebook_user = @profile_user.get_value_of_power_tag("facebook")
+        @instagram_user = @profile_user.get_value_of_power_tag("instagram")
+
```

Comment:
@Forchapeatl `User#social_link` calls `User#get_value_of_power_tag` internally so it’s kind of inefficient to be calling it twice. What you could do is remove the `@facebook`, `@twitter` e.t.c variables and use only the `@github_user` e.t.c variables.

After that we can add the `social_links` method as a view helper

Reply:
@TildaDares In regards to moveing the `social_links` method
of  [https://github.com/publiclab/plots2/blob/main/app/helpers/user_tags_helper.rb](https://github.com/publiclab/plots2/blob/main/app/helpers/user_tags_helper.rb) please is this what is requried , I replace the `UserTag` by `social_links`  ?
```suggestion
module UserTagsHelper
  def fetch_tags(uid, type)
    tag_types = %w(skill gear role tool)
    tags = []
    if tag_types.include? type
      tags = social_links.where(uid: uid).where('value LIKE ?', type + ':' + '%')
    end
    tags
  end
end
```

Reply:
No I mean move `social_links` and add it as another method in the `user_tags` file

Reply:
- Navigate to the file https://github.com/publiclab/plots2/blob/main/app/helpers/user_tags_helper.rb
- On line 10 add the following lines of code

def social_link(site, user_name)
    "https://#{site}.com/#{user_name}"
  end

Reply:
Also, if something I said isn’t clear enough, please feel free to tell me and I’ll do my best to explain in an east to understand way. I’m here to help you.

Reply:
Also, if something I said isn’t clear enough, please feel free to tell me and I’ll do my best to explain in an east to understand way. I’m here to help you.

Reply:
@TildaDares , I'm sorry I  feared I will be viewed as incompetent if I ask too many questions.
I did as instructed but the social icons  seem not to show up.
![image](https://user-images.githubusercontent.com/24577149/159113308-dea4e21d-0ef7-4a92-b9b6-83bd001b03af.png)
Some information 
- user_tags_helper.rb
![image](https://user-images.githubusercontent.com/24577149/159113336-a2e91353-e122-4047-a1b3-b55c0a67fa91.png)
- users_controller.rb
![image](https://user-images.githubusercontent.com/24577149/159113379-82ec85fe-989a-46dd-aef5-e7bd1a9cce24.png)

Reply:
> @TildaDares , I'm sorry I feared I will be viewed as incompetent if I ask too many questions.

It's a good thing to ask a lot of questions.

Reply:
@Forchapeatl Can you push those changes to this branch? I can't really tell what the problem is from the screenshot.

Reply:
@TildaDares I have merged the changes to` patch-2`

Reply:
@TildaDares  It is done , I forgot to replace `<% if @github.nil? == false %>` by `<% if @github_user.nil? == false %>`  of plots2/app/views/profile.html.erb file. The UI now  loads as expected   please have a look at  [http://194.163.138.111/profile/user](http://194.163.138.111/profile/user). I send gratitude .

Reply:
Great job @Forchapeatl! You can now remove the social_links method from `user.rb`.

Reply:
![Screenshot_20220321-093525](https://user-images.githubusercontent.com/24577149/159227621-8126d1c8-0c82-4555-8aae-edd8afc5c1c4.png)
@TildaDares  thank you. I have removed the social_links method as requested.

Reply:
@TildaDares  I have removed the social_links method from [users_controller.rb](https://github.com/publiclab/plots2/pull/10788/commits/3a43c90009e2a2ee3855668086c35c583b60cd32) as requested.

Reply:
@TildaDares , It is done. Thank you.